src/vsr/replica_test.zig

```diff
@@ -835,6 +835,51 @@ test "Cluster: view-change: DVC, 2/3 faulty header stall" {
     try expectEqual(t.replica(.R_).status(), .view_change);
 }
 
+test "Cluster: view-change: duel of the primaries" {
+    // In a cluster of 3, one replica gets partitioned away, and the remaining two _both_ become
+    // primaries (for different views). Additionally, the primary from the  higher view is
+    // abdicating. The primaries should figure out that they need to view-change to a higher view.
+    const t = try TestContext.init(.{ .replica_count = 3 });
+    defer t.deinit();
+
+    var c = t.clients(0, t.cluster.clients.len);
+    try c.request(20, 20);
+    try expectEqual(t.replica(.R_).commit(), 20);
+
+    try expectEqual(t.replica(.R_).view(), 1);
+    try expectEqual(t.replica(.R1).role(), .primary);
+
+    t.replica(.R2).drop_all(.R_, .bidirectional);
+    t.replica(.R1).drop(.R_, .outgoing, .commit);
+    try c.request(21, 21);
+
+    try expectEqual(t.replica(.R0).commit_max(), 20);
+    try expectEqual(t.replica(.R1).commit_max(), 21);
+    try expectEqual(t.replica(.R2).commit_max(), 20);
+
+    t.replica(.R0).pass_all(.R_, .bidirectional);
+    t.replica(.R2).pass_all(.R_, .bidirectional);
+    t.replica(.R1).drop_all(.R_, .bidirectional);
+    t.replica(.R2).drop(.R0, .bidirectional, .prepare_ok);
+    t.replica(.R2).drop(.R0, .outgoing, .do_view_change);
+    t.run();
+
+    // The stage is set: we have two primaries in different views, R2 is about to abdicate.
+    try expectEqual(t.replica(.R1).view(), 1);
+    try expectEqual(t.replica(.R1).role(), .primary);
+    try expectEqual(t.replica(.R1).commit(), 21);
+    try expectEqual(t.replica(.R2).role(), .primary);
+    try expectEqual(t.replica(.R2).commit(), 20);
```

Comment:
```suggestion
    // The stage is set: we have two primaries in different views, R2 is about to abdicate.
    try expectEqual(t.replica(.R1).view(), 1);
    try expectEqual(t.replica(.R1).status(), .normal);
    try expectEqual(t.replica(.R1).role(), .primary);
    try expectEqual(t.replica(.R1).commit(), 21);
    try expectEqual(t.replica(.R2).status(), .normal);
    try expectEqual(t.replica(.R2).role(), .primary);
    try expectEqual(t.replica(.R2).commit(), 20);
    try expectEqual(t.replica(.R2).op_head(), 21);
```

The `status=normal` and `op_head > commit_min` assertion emphasizes/checks that R2 will indeed abdicate.

Reply:
Ah, right, `.role() == .primary` doesn't imply `.normal`, unlike the `replica.primary()` function!

Reply:
And I also forgot to assert R2's view!